ddtrace/contrib/openai/patch.py

```diff
@@ -0,0 +1,456 @@
+        span.set_tag_str("api_base", openai.api_base)
+    if hasattr(openai, "api_version") and openai.api_version:
+        span.set_tag_str("api_version", openai.api_version)
+    if hasattr(openai, "organization") and openai.organization:
+        span.set_tag_str("organization", openai.organization)
+
+
+_completion_request_attrs = [
+    "model",
+    "suffix",
+    "max_tokens",
+    "temperature",
+    "top_p",
+    "n",
+    "stream",
+    "logprobs",
+    "echo",
+    "stop",
+    "presence_penalty",
+    "frequency_penalty",
+    "best_of",
+    "logit_bias",
+    "user",
+    "prompt",
+]
+
+
+def _completion_create(openai, pin, instance, args, kwargs):
+    model = kwargs.get("model")
+    span = pin.tracer.trace(
+        "openai.request", resource="completions/%s" % model, service=trace_utils.ext_service(pin, config.openai)
+    )
+    _init_openai_span(span, openai)
+    if model:
+        span.set_tag_str("model", model)
+
+    sample_pc_span = _is_pc_sampled_span(span)
+    sample_pc_log = _is_pc_sampled_log(span)
+
+    prompt = kwargs.get("prompt")
+    if sample_pc_span:
+        if isinstance(prompt, list):
+            for idx, p in enumerate(prompt):
+                span.set_tag_str("request.prompt.%d" % idx, _process_text(p))
+        else:
+            span.set_tag_str("request.prompt", _process_text(prompt))
+
+    for kw_attr in _completion_request_attrs:
+        if kw_attr in kwargs:
+            if kw_attr != "prompt":
+                span.set_tag("request.%s" % kw_attr, kwargs[kw_attr])
+
+    resp, error = yield span
+
+    metric_tags = [
+        "model:%s" % kwargs.get("model"),
+        "endpoint:%s" % instance.OBJECT_NAME,
+        "error:%d" % (1 if error else 0),
+    ]
+    if openai.organization:
+        metric_tags.append("organization:%s" % openai.organization)
+
+    if error is not None:
+        span.set_exc_info(*sys.exc_info())
+        stats_client().increment("error", 1, tags=metric_tags + ["error_type:%s" % error.__class__.__name__])
+        _dd_log(
+            span,
+            "error",
+            "openai error",
+            tags=metric_tags,
+            attrs={
+                "prompt": prompt,
+            },
+        )
+    if resp and not kwargs.get("stream"):
+        if "choices" in resp:
+            choices = resp["choices"]
+            span.set_tag("response.choices.num", len(choices))
+            for choice in choices:
+                idx = choice["index"]
+                if "finish_reason" in choice:
+                    span.set_tag_str("response.choices.%d.finish_reason" % idx, str(choice["finish_reason"]))
+                if "logprobs" in choice:
+                    span.set_tag("response.choices.%d.logprobs" % idx, choice["logprobs"])
+                if sample_pc_span:
+                    span.set_tag("response.choices.%d.text" % idx, _process_text(choice.get("text")))
+        span.set_tag("response.id", resp["id"])
```

Comment:
Is there a use case for this response id?

Reply:
I think the idea is that it can be provided back to OpenAI if there is something about the response that is worth talking to their support about.

Reply:
we opted to remove it since this would be a pretty exceptional use-case